src/node_url_pattern.cc

```diff
@@ -0,0 +1,791 @@
+#include "node_url_pattern.h"
+#include "base_object-inl.h"
+#include "debug_utils-inl.h"
+#include "env-inl.h"
+#include "node.h"
+#include "node_errors.h"
+#include "node_mem-inl.h"
+#include "path.h"
+#include "util-inl.h"
+
+namespace node::url_pattern {
+
+using v8::Array;
+using v8::Context;
+using v8::DontDelete;
+using v8::FunctionCallbackInfo;
+using v8::FunctionTemplate;
+using v8::Global;
+using v8::Isolate;
+using v8::Local;
+using v8::LocalVector;
+using v8::MaybeLocal;
+using v8::Name;
+using v8::NewStringType;
+using v8::Object;
+using v8::PropertyAttribute;
+using v8::ReadOnly;
+using v8::RegExp;
+using v8::String;
+using v8::Value;
+
+std::optional<URLPatternRegexProvider::regex_type>
+URLPatternRegexProvider::create_instance(std::string_view pattern,
+                                         bool ignore_case) {
+  auto isolate = Isolate::GetCurrent();
+  DCHECK_NOT_NULL(isolate);
+  auto env = Environment::GetCurrent(isolate);
+  int flags = RegExp::Flags::kUnicodeSets;
+  if (ignore_case) {
+    flags |= static_cast<int>(RegExp::Flags::kIgnoreCase);
+  }
+
+  Local<String> local_pattern;
+  if (!String::NewFromUtf8(
+           isolate, pattern.data(), NewStringType::kNormal, pattern.size())
+           .ToLocal(&local_pattern)) {
+    return std::nullopt;
+  }
+  Local<RegExp> regexp;
+  if (!RegExp::New(
+           env->context(), local_pattern, static_cast<RegExp::Flags>(flags))
+           .ToLocal(&regexp)) {
+    return std::nullopt;
+  }
+  return Global<RegExp>(isolate, regexp);
+}
+
+bool URLPatternRegexProvider::regex_match(std::string_view input,
+                                          const regex_type& pattern) {
+  auto isolate = Isolate::GetCurrent();
+  DCHECK_NOT_NULL(isolate);
```

Comment:
nit: See https://github.com/nodejs/node/pull/56452/files#r1933001727, this DCHECK is not necessary.

Reply:
Thanks. Removing.